xarray/core/utils.py

```diff
@@ -489,3 +490,13 @@ def ensure_us_time_resolution(val):
     elif np.issubdtype(val.dtype, np.timedelta64):
         val = val.astype('timedelta64[us]')
     return val
+
+
+@functools.partial(np.vectorize, otypes='O')
+def encode_pickle(obj):
+    return np.frombuffer(pickle.dumps(obj), dtype=np.uint8)
```

Comment:
The way that pickle works, any version of Python can load older pickles, but old versions of Python can't load newer pickles. So `protocol=2` is a maximally backwards compatible option, but misses out on any later pickle improvements.